src/main/java/org/wso2/securevault/secret/SecretManager.java

```diff
@@ -100,71 +132,33 @@ public void init(Properties properties) {
             }
         }
 
-        String repositoriesString = MiscellaneousUtil.getProperty(
-                configurationProperties, PROP_SECRET_REPOSITORIES, null);
-        if (repositoriesString == null || "".equals(repositoriesString)) {
-            if (log.isDebugEnabled()) {
-                log.debug("No secret repositories have been configured");
-            }
-            return;
-        }
-
-        String[] repositories = repositoriesString.split(",");
-        if (repositories == null || repositories.length == 0) {
-            if (log.isDebugEnabled()) {
-                log.debug("No secret repositories have been configured");
-            }
-            return;
-        }
-
-
-        //Create a KeyStore Information  for private key entry KeyStore
-        IdentityKeyStoreInformation identityInformation =
-                KeyStoreInformationFactory.createIdentityKeyStoreInformation(properties);
-
-        // Create a KeyStore Information for trusted certificate KeyStore
-        TrustKeyStoreInformation trustInformation =
-                KeyStoreInformationFactory.createTrustKeyStoreInformation(properties);
-
-        String identityKeyPass = null;
-        String identityStorePass = null;
-        String trustStorePass = null;
-        if(identityInformation != null){
-            identityKeyPass = identityInformation
-                    .getKeyPasswordProvider().getResolvedSecret();
-            identityStorePass = identityInformation
-                    .getKeyStorePasswordProvider().getResolvedSecret();
-        }
-
-        if(trustInformation != null){
-            trustStorePass = trustInformation
-                .getKeyStorePasswordProvider().getResolvedSecret();
-        }
-
+        getAllProviders(configurationProperties);
 
-        if (!validatePasswords(identityStorePass, identityKeyPass, trustStorePass)) {
-            if (log.isDebugEnabled()) {
-                log.info("Either Identity or Trust keystore password is mandatory" +
-                        " in order to initialized secret manager.");
-            }
+        if (!isSecureVaultStatusValidated()) {
             return;
         }
 
         IdentityKeyStoreWrapper identityKeyStoreWrapper = new IdentityKeyStoreWrapper();
-        identityKeyStoreWrapper.init(identityInformation, identityKeyPass);
-
         TrustKeyStoreWrapper trustKeyStoreWrapper = new TrustKeyStoreWrapper();
-        if(trustInformation != null){
-            trustKeyStoreWrapper.init(trustInformation);            
+        if (legacyProvidersExist) {
+            if (!createKeyStoreWrappers(identityKeyStoreWrapper, trustKeyStoreWrapper, properties)) {
```

Comment:
createKeyStoreWrappers cannot return a boolean. I think you can use this as a validation method and do the wrapping part in another private method

Reply:
Modified